bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/discovery/ThingDiscoveryService.java

```diff
@@ -0,0 +1,243 @@
+/**
+ * Copyright (c) 2010-2023 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.boschshc.internal.discovery;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+import org.apache.commons.lang3.StringUtils;
```

Comment:
Sorry for such late finding, was just following up on https://github.com/openhab/openhab-addons/pull/14197#discussion_r1141044386 and noticed this. We are actually trying to get rid of all usages of this class. See #7722 and a list PR's using the filter "is:open is:pr in:title org.apache.common".

Reply:
No problem, but I'm not able provide a last minute fix today, because it took me some hours to find a solution.
I will replace the used ``StringUtils.splitByCharacterTypeCamelCase`` function during the next days

Reply:
Sure, let's not rush it, it would be counter-productive to introduce more usages while trying to get rid of those we already have. 🙂 I don't have any further comments, so we should be able to merge shortly after refactoring this.

Reply:
You could use https://www.baeldung.com/java-camel-case-title-case-to-words and/or https://stackoverflow.com/questions/7593969/regex-to-split-camelcase-or-titlecase-advanced as a starting point.

Reply:
And also of course: https://github.com/apache/commons-lang/blob/2cf6486197b268bac818d05e38173ab57ce16315/src/main/java/org/apache/commons/lang3/StringUtils.java#L7458-L7460

Perhaps only a subset of that functionality is needed.

Reply:
I took the org.apache.commons.lang code as an example to implement own code to add spaces for camel cases.

Reply:
Sorry, way to late as it is allready merged. Just want to mention an implementation was available, including tests, here:  https://github.com/openhab/openhab-addons/pull/14434/files